### PR TITLE
feat(tablekit-react): add toggle component

### PR DESCRIPTION
**Pull request by irmir**
Completes #105

Screenshot 2022-10-13 at 22 20 21

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @tablecheck/tablekit-css@2.1.0-canary.129.3428418383.0
  npm install @tablecheck/tablekit-react-select@2.1.0-canary.129.3428418383.0
  npm install @tablecheck/tablekit-react@2.1.0-canary.129.3428418383.0
  # or 
  yarn add @tablecheck/tablekit-css@2.1.0-canary.129.3428418383.0
  yarn add @tablecheck/tablekit-react-select@2.1.0-canary.129.3428418383.0
  yarn add @tablecheck/tablekit-react@2.1.0-canary.129.3428418383.0
  ```
</details>

### CODE REVIEW

**Review by SimeonC — CHANGES_REQUESTED** [review]
CleanShot 2022-09-30 at 15 51 25

Missing indeterminate state values, also size (see component and slack messages about that).

https://css-tricks.com/almanac/selectors/i/indeterminate/

**Comment by irmir** [comment]
Thank you, done

**Comment by SimeonC** [comment]
Turned on the "keep branches up to date" check - can you rebase and then I can check the Chromatic build.

File: system/react/src/components/Toggle.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+          disabled={variant === 'Disabled'}
+          ref={refs1.current[index]}
+        />
+      ))}
+      {contentVariants.map((variant, index) => (
+        <Toggle
+          data-pseudo={variant.toLowerCase()}
+          data-size="small"
+          disabled={variant === 'Disabled'}
+          ref={refs2.current[index]}
```
Comment (SimeonC) [inline system/react/src/components/Toggle.stories.tsx:65]: This is simpler and needs less code.
```suggestion
          ref={(ref) => { if (ref) ref.indeterminate = true; }}
```

**Review by SimeonC — CHANGES_REQUESTED** [review]

File: system/react/src/components/Toggle.stories.tsx (hunk 1 above)
Comment (irmir) [reply system/react/src/components/Toggle.stories.tsx]: thank you! done

**Review by SimeonC — APPROVED** [review]
Nice work!

**Review by SimeonC — COMMENTED** [review]
Let's hold off on merging this a little bit - we don't have focus styles in figma to finish off the design.

**Review by SimeonC — CHANGES_REQUESTED** [review]
CleanShot 2022-10-07 at 19 47 38

Note that transparent border is wrong - it should be a darker purple. We also have the focus styles in figma so we can complete that.

**Comment by irmir** [comment]
> Note that transparent border is wrong - it should be a darker purple. We also have the focus styles in figma so we can complete that.

@SimeonC I've added focus state but not sure about purple border. It uses transparent border in Figma not darker purple:
Screenshot 2022-10-13 at 21 53 31

File: system/react/src/components/Toggle.ts [hunk 2]
```diff
@@ -0,0 +1,121 @@
+import styled from '@emotion/styled';
+
+export const classySelector = 'input[type="checkbox"].toggle';
+
+export const Toggle = styled.input<{ type?: 'checkbox' }>`
```
Comment (SimeonC) [inline system/react/src/components/Toggle.ts:5]:
```suggestion
export const Toggle = styled.input<{ type: 'checkbox' }>`
```

This styling is only applicable to a checkbox so we should enforce it.

**Review by SimeonC — CHANGES_REQUESTED** [review]

**Comment by SimeonC** [comment]
> @SimeonC I've added focus state but not sure about purple border. It uses transparent border in Figma not darker purple:

Yes that's correct, there were some cases in the last version where the border was `transparent` - literally you could see the background colour through it.

This version looks correct.

File: system/react/src/components/Toggle.ts (hunk 2 above)
Comment (irmir) [reply system/react/src/components/Toggle.ts]: fixed

**Comment by irmir** [comment]
@SimeonC looks like everything is ok in PR now, can you please have a look at audit checks?

**Review by SimeonC — APPROVED** [review]

**Review by SashaShostyr — APPROVED** [review]

**Pull request merged**